sympy/combinatorics/tests/test_coset_table.py

```diff
@@ -707,3 +707,110 @@ def test_look_ahead():
     C_c = coset_enumeration_c(f, H)
     C_c.compress(); C_c.standardize()
     assert C_c.table[: 10] == table0
+
+def test_modified_methods():
+    '''
+    Tests for modified coset table methods.
+    Example 5.7 from [1] Holt, D., Eick, B., O'Brien
+    "Handbook of Computational Group Theory".
+
+    '''
```

Comment:
I think the modified tests should have an actual coset enumeration test - something that would catch the error that you've just fixed (the one with the infinite loop)